pkg/prometheus/statefulset.go

```diff
@@ -70,6 +71,7 @@ var (
 	}
 	shardLabelName                = "operator.prometheus.io/shard"
 	prometheusNameLabelName       = "operator.prometheus.io/name"
+	http2AnnotationName           = "operator.prometheus.io/http2"
```

Comment:
I assume that we can avoid the extra annotation if we had the http2 bool to the `hashstructure.Hash` struct in [createSSetInputHash](https://github.com/prometheus-operator/prometheus-operator/blob/69f70617961b839bd34f1ff601c3653b20000091/pkg/prometheus/operator.go#L1707)?